plugins/package-managers/go/src/main/kotlin/GoMod.kt

```diff
@@ -484,3 +485,28 @@ internal fun parseWhyOutput(output: String): Set<String> {
 
     return usedModules
 }
+
+/**
+ * Module paths appear as substrings of file system paths in the module cache on the file system.
+ * Go does not rely on the file system to be case-sensitive. For this reason, Go has decided to
+ * replace every uppercase letter in file system paths with an exclamation mark followed by the
+ * letter's lowercase equivalent.
+ *
+ * Details behind the reasoning and implementation in Go can be found in the Go source code at
+ * [module.go](https://github.com/golang/go/blob/5b6d3dea8744311825fd544a73edb8d26d9c7e98/src/cmd/vendor/golang.org/x/mod/module/module.go#L33-L42C64)
+ */
+internal fun escapeVersion(version: String): String {
+    require("!" !in version) { "Module versions must not contain exclamation marks: $version"}
+
+    val escaped = buildString {
```

Comment:
Nit: You can omit the `escaped` variable and directly return from here: `return buildString {`.

Reply:
Done.